vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -39,10 +39,10 @@ def save_inventory(type, parent, hash)
     # Find the record, and update if found, else create it
     found = parent.send(type)
     if found.nil?
-      found = parent.send("build_#{type}", hash)
+      found = parent.send("build_#{type}", hash.except(:id))
```

Comment:
I'm not sure about each of these four lines individually... at least one specifically needs it to make a test pass. At that point, it seemed most sane/safe to apply it across the board: no matter what, it would be a very bad idea for an ID to end up in any of these.